### ISSUE TITLE
Add support for with snapshot()

### ISSUE THREAD

**Issue by zmievsa** [15r10nk/inline-snapshot#81]
When you develop libraries, you often need to check that your library raises some exception at runtime.

I created better-devtools (Samuel Colvin's devtools fork) to [automate this process](https://github.com/zmievsa/better-devtools/blob/7e4961722d34ec0036b3b8c45e962f59086169c4/devtools/pytest_plugin.py#L79) and I would like to propose a similar tool for snapshot.

It would allow for something like the following syntax:

```python
with snapshot():
    raise ValueError
```

Which would translate into something like the following:

```python
with snapshot(pytest.raises(ValueError)):
    raise ValueError
```

**Comment by 15r10nk** [comment]
I use something like this in my own tests.

```python
def test_with():
    with raises(snapshot("ZeroDivisionError: division by zero")):
        print(1/0)
```

the implementation looks like this:

```python
@contextlib.contextmanager
def raises(snapshot):
    with pytest.raises(Exception) as excinfo:
        yield

    assert f"{type(excinfo.value).__name__}: {excinfo.value}" == snapshot
```

would this fit your needs? I could put such generic utils in a `inline_snapshot.extras` module.

**Comment by zmievsa** [comment]
Yup, it would! Looks great! You really did think of all use cases. 

This is an amazing project.

**Comment by alexmojaki** [comment]
That is very cool. I think I've had times where I would have used this, and it didn't occur to me that vanilla `snapshot` could do the trick.

**Comment by 15r10nk** [comment]
this was implemented in 0.12.0 https://15r10nk.github.io/inline-snapshot/extra/#inline_snapshot.extra.raises

**Comment by zmievsa** [comment]
You're a rockstar! Thanks!
